prime-router/src/main/kotlin/azure/RequestFunction.kt

```diff
@@ -50,14 +50,24 @@ abstract class RequestFunction(
             ?: request.queryParameters.getOrDefault(CLIENT_PARAMETER, "")
     }
 
+    class InvalidExternalPayloadException(val externalName: String) : RuntimeException()
+
     /**
      * Extract the optional payloadName (aka sender-supplied filename) from request headers or query string parameters
      * @param request the http request message from the client
      */
-    protected fun extractPayloadName(request: HttpRequestMessage<String?>): String? {
+    internal fun extractPayloadName(request: HttpRequestMessage<String?>): String? {
         // payloadName can be in the header or in the url parameters.  Return null if not found.
-        return request.headers[PAYLOAD_NAME_PARAMETER]
+        val payloadName = request.headers[PAYLOAD_NAME_PARAMETER]
             ?: request.queryParameters[PAYLOAD_NAME_PARAMETER]
+
+        payloadName?.let {
+            if (it.length > ReportFile.REPORT_FILE.EXTERNAL_NAME.dataType.length()) {
```

Comment:
Does `ReportFile.REPORT_FILE.EXTERNAL_NAME.dataType.length() == 2048` because the column is a `varchar(2048)`?

Reply:
Yes that is correct